Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs

```diff
@@ -0,0 +1,354 @@
+            var internalRequest = internalRequestAsResource as TEntityInterface;
+
+            if (request != null)
+            {
+                MapAll(request, internalRequest);
+            }
+
+            //TODO: Add support for If-None-Match; current implementation cannot verify value without going to the db
+            // Read the incoming ETag header, if present
+            Request.TryGetRequestHeader(HeaderConstants.IfNoneMatch, out string etagValue);
+
+            var queryParameters = new QueryParameters(urlQueryParametersRequest);
+
+            // Execute the pipeline (synchronously)
+            var result = await GetManyPipeline.Value
+                .ProcessAsync(
+                    new GetManyContext<TResourceReadModel, TAggregateRoot>(internalRequestAsResource, queryParameters),
+                    new CancellationToken());
+
+            // Handle exception result
+            if (result.Exception != null)
+            {
+                Logger.Error(GetAllRequest, result.Exception);
+                return CreateActionResultFromException(result.Exception);
+            }
+
+            // Return multiple results
+            if (queryParameters.TotalCount)
+            {
+                Response.Headers.Add(HeaderConstants.TotalCount, result.ResultMetadata.TotalCount.ToString());
+            }
+
+            return Ok(result.Resources);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status412PreconditionFailed)]
+        public virtual async Task<IActionResult> Get(Guid id)
+        {
+            // Read the incoming ETag header, if present
+            Request.TryGetRequestHeader(HeaderConstants.IfNoneMatch, out string etagValue);
+
+            // Execute the pipeline (synchronously)
+            var result = await GetByIdPipeline.Value.ProcessAsync(
+                new GetContext<TAggregateRoot>(id, etagValue), CancellationToken.None);
+
+            // Handle exception result
+            if (result.Exception != null)
+            {
+                Logger.Error(GetByIdRequest, result.Exception);
+                return CreateActionResultFromException(result.Exception);
+            }
+
+            // Handle success result
+            // Add ETag header for the resource
+            Response.Headers[HeaderConstants.ETag] = GetEtag(result.Resource.ETag).ToString();
+
+            return Ok(result.Resource);
+        }
+
+        [CheckModelForNull]
+        [HttpPut("{id}")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status412PreconditionFailed)]
+        public virtual async Task<IActionResult> Put([FromBody] TPutRequest request, [FromQuery] Guid id)
+        {
+            // Manual binding of Id to main request model
```

Comment:
We lost ModelState.IsValid checks here (as added in https://github.com/Ed-Fi-Alliance/Ed-Fi-ODS/commit/845bae2ce7470e91a957222be4fd108b463b280d).  Is this unnecessary in .net core or are we regressing?

Reply:
looks like i missed that change. I will create a ticket. ODS-4494